### ISSUE TITLE
Linux: reload window moves tab labels up a bit for me

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#76573]
Since I do not see this in insiders, possibly a regression from the icon work:

* open VSCode on Linux
* open some tabs
* reload window

=> :bug: tab labels pushed up a little bit

![Peek 2019-07-04 10-48](https://user-images.githubusercontent.com/900690/60652883-43466980-9e49-11e9-8bd2-bc082dc77fa4.gif)

//cc @isidorn

**Comment by isidorn** [comment]
Look closely in the gif to ther right = they get pushed because an icon shows up in your editor title with a slight delay.
This icon is probably coming from an extension.
I think this is as designed, you just do no see it anywhere else because you do not have that extension.

**Comment by bpasero** [comment]
GOod catch, let me check that..

**Comment by bpasero** [comment]
@isidorn caused by https://github.com/microsoft/vscode/commit/c0a9bc3988960a0edcb262ced7cc4793503a2df5 (after bisect)

**Comment by isidorn** [comment]
Sorry that you had to bisect, I had that commit as a suspect.

What was the previous behavior? Because what I changed is that the ResourceLabel gets re-rendered when a new formatter gets registered. 
When they get re-rendered they seem to make up space for the new icon.
This all looks good to me.

**Comment by bpasero** [comment]
@isidorn the previous behaviour was that the label was not moving up/down. I can only reproduce on Linux btw, so maybe check there.

**Comment by bpasero** [comment]
@isidorn I am no longer able to reproduce https://github.com/microsoft/vscode/issues/76209 after the [change](https://github.com/microsoft/vscode/commit/1b16063cca17e19af49dfd006c400ec360b9d113) I just pushed that simplifies the code a bit. Specifically:
* no longer calling `this.setFile` from label anymore but just `render`
* slightly unrelated: making sure we are not clearing the icon cache when the path label changes because only a resource or file type change can trigger that

Bottom line: I think the memoizing of labels in file editor input was the only reason this happened. Any reason you added the `this.setFile` in the first place? Note that `this.setFile` will trigger a `render` automatically, so we always called this twice so far.

**Comment by isidorn** [comment]
@bpasero looks like a good change. If you can not repro https://github.com/microsoft/vscode/issues/76209 anymore then we are good.
I do not remember the exact reason why I added `this.setFile`, my guess would be that it just worked and I was happy to keep it and I was unaware of the double render.
